src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -7825,6 +7860,327 @@ private static int Test(int x, out int y)
+                Assert.Equal(SymbolInfo.None, symbolInfo);
+
+                symbolInfo = model.GetSpeculativeSymbolInfo(baseWithargs.ArgumentList.OpenParenToken.SpanStart, (SyntaxNode)speculativeBaseInitializer, SpeculativeBindingOption.BindAsExpression);
+                Assert.Equal("Base..ctor(System.Int32 X)", symbolInfo.Symbol.ToTestDisplayString());
+
+                symbolInfo = CSharpExtensions.GetSpeculativeSymbolInfo(model, baseWithargs.ArgumentList.OpenParenToken.SpanStart, speculativeBaseInitializer);
+                Assert.Equal("Base..ctor(System.Int32 X)", symbolInfo.Symbol.ToTestDisplayString());
+
+                Assert.Equal(TypeInfo.None, model.GetSpeculativeTypeInfo(baseWithargs.ArgumentList.OpenParenToken.SpanStart, (SyntaxNode)speculativePrimaryInitializer, SpeculativeBindingOption.BindAsExpression));
+            }
+        }
+
+        [Fact]
+        public void BaseArguments_20()
+        {
+            var src = @"
+class Base
+{
+    public Base(int X)
+    {
+    }
+
+    public Base() {}
+}
+
+class C : Base(GetInt(X, out var xx) + xx, Y), I
+{
+    C(int X, int Y, int Z) : base(X, Y, Z, 1) { return; }
+
+    static int GetInt(int x1, out int x2)
+    {
+        throw null;
+    }
+}
+
+interface I {}
+";
+
+            var comp = CreateCompilation(src);
+
+            comp.VerifyDiagnostics(
+                // (11,15): error CS8861: Unexpected argument list.
+                // class C : Base(GetInt(X, out var xx) + xx, Y), I
+                Diagnostic(ErrorCode.ERR_UnexpectedArgumentList, "(").WithLocation(11, 15),
+                // (13,30): error CS1729: 'Base' does not contain a constructor that takes 4 arguments
+                //     C(int X, int Y, int Z) : base(X, Y, Z, 1) { return; }
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "base").WithArguments("Base", "4").WithLocation(13, 30)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            SymbolInfo symbolInfo;
+            PrimaryConstructorBaseTypeSyntax speculativePrimaryInitializer;
+            ConstructorInitializerSyntax speculativeBaseInitializer;
+
+            {
+                var baseWithargs = tree.GetRoot().DescendantNodes().OfType<PrimaryConstructorBaseTypeSyntax>().Single();
+                Assert.Equal("Base(GetInt(X, out var xx) + xx, Y)", baseWithargs.ToString());
+                Assert.Equal("Base", model.GetTypeInfo(baseWithargs.Type).Type.ToTestDisplayString());
+                Assert.Equal(TypeInfo.None, model.GetTypeInfo(baseWithargs));
+                symbolInfo = model.GetSymbolInfo((SyntaxNode)baseWithargs);
+                Assert.Equal(SymbolInfo.None, symbolInfo);
+                symbolInfo = model.GetSymbolInfo(baseWithargs);
+                Assert.Equal(SymbolInfo.None, symbolInfo);
```

Comment:
Why don't we get a list of failed candidates here?

Reply:
> Why don't we get a list of failed candidates here?

Because there are no candidates, the PrimaryConstructorBaseTypeSyntax is never bound for anything other than a record.  

---
In reply to: [444534737](https://github.com/dotnet/roslyn/pull/45351#discussion_r444534737) [](ancestors = 444534737)